service.go

```diff
@@ -204,22 +205,33 @@ type serviceManager struct {
 	// the onet host
 	server *Server
 	// a bbolt database for all services
-	db *bolt.DB
+	db     *bolt.DB
+	dbPath string
+	// should the db be deleted on close?
+	delDb bool
 	// the dispatcher can take registration of Processors
 	network.Dispatcher
 }
 
-const configFolder = "config"
-
 // newServiceManager will create a serviceStore out of all the registered Service
-func newServiceManager(c *Server, o *Overlay) *serviceManager {
-	db, err := openDb(c.dbFileName())
+func newServiceManager(svr *Server, o *Overlay, dbPath string) *serviceManager {
+	services := make(map[ServiceID]Service)
+	s := &serviceManager{
+		services:   services,
+		server:     svr,
+		dbPath:     dbPath,
+		Dispatcher: network.NewRoutineDispatcher(),
+	}
+	if dbPath == "" {
+		s.delDb = true
```

Comment:
I'm nitpicking a bit - if we already have `delDb`, we don't need to check for `dbPath` anymore, right? `newServiceManager` and `newService` would take an additional parameter for `delDb`.

Reply:
ops, `newServer`, not `newService`